src/Datadog.Trace/Span.cs

```diff
@@ -264,6 +270,17 @@ public void Finish(DateTimeOffset finishTimestamp)
             if (shouldCloseSpan)
             {
                 Context.TraceContext.CloseSpan(this);
+                if (Log.IsEnabled(LogEventLevel.Debug))
```

Comment:
Is there a reason this logging statement is hidden under the `Log.IsEnabled` check whereas the span start isn't?

Reply:
To prevent the interpolation. Happy to refactor it otherwise.

We have a bunch of spots in c++ that do this on that note.